lib/plugins/aws/deploy/lib/checkForChanges.js

```diff
@@ -187,7 +187,6 @@ module.exports = {
    */
   async checkLogGroupSubscriptionFilterResourceLimitExceeded() {
     const region = this.provider.getRegion();
-    const cloudWatchLogsSdk = new this.provider.sdk.CloudWatchLogs({ region });
```

Comment:
This PR should be only about tests refactor. Let's not introduce changes to this file

Reply:
I had a look again, and I'm not sure how I should stub the sdk.CloudwatchLogs with runServerless. From what I understand, awsRequestStubMap only works for to replace request in the provider. Is there a way to stub the provider.sdk without doing it "manually"?

Reply:
Indeed @AlinoeDoctari, I totally missed that. Ideally if stub somehow stubs `aws-sdk` directly, still to not make a dependency on that here, let's keep this refactor here as you proposed initially.